src/types/type.ts

```diff
@@ -89,7 +89,9 @@ export abstract class ComplexType<S, T> implements IType<S, T> {
     }
 
     reconcile(current: Node, newValue: any): Node {
-        const { parent, subpath } = current
+        if (current.snapshot === newValue)
```

Comment:
A: This is always called before preProcess, as this check is performed in node "applySnapshot", and after that the type applySnapshot kicks in, so should be everything fine  :) I can't think of issues too, maybe will just end up skipping always the perf improvement ;)